spiketoolkit/validation/metric_calculator.py

```diff
@@ -79,26 +87,34 @@ def __init__(self, sorting, recording=None, sampling_frequency=None, unit_ids=No
                                     end_frame=epoch[2] * self._sampling_frequency)
         if recording is not None:
             assert isinstance(recording, RecordingExtractor), "'sorting' must be  a RecordingExtractor object"
-            self.set_recording(recording)
+            self.set_recording(recording, apply_filter=apply_filter, freq_min=freq_min, freq_max=freq_max)
         else:
             self._recording = None
 
-    def set_recording(self, recording):
+    def set_recording(self, recording, apply_filter=True, freq_min=300, freq_max=6000):
         '''
         Sets given recording extractor
 
         Parameters
         ----------
         recording: RecordingExtractor
             The recording extractor to be stored.
+        apply_filter: bool
+            If True, recording is bandpass-filtered.
+        freq_min: float
+            High-pass frequency for optional filter (default 300 Hz).
+        freq_max: float
+            Low-pass frequency for optional filter (default 6000 Hz).
         '''
+        if apply_filter:
+            recording = st.preprocessing.bandpass_filter(recording=recording, freq_min=freq_min, freq_max=freq_max, cache=True)
```

Comment:
Here I would do
```python
self._recording = recording
if apply_filter:
    self._fileterd_recording = st.preprocessing.bandpass_filter(recording=recording,
             freq_min=freq_min, freq_max=freq_max, cache=True)
else:
    self._fileterd_recording = recording
```

Reply:
I can set a bool=is_filtered

Reply:
What do you think @samuelgarcia ? Do I keep it the same, add a bool, or do what you said? I would keep it the same or add a bool personally

Reply:
Added the bool